server/s3_motr_kvs_writer.cc

```diff
@@ -599,7 +600,7 @@ int S3MotrKVSWriter::put_keyval_impl(
 
     // Operation callbacks are used in async mode.
     if (idx_op_ctx->cbs->oop_executed) {
```

Comment:
I think, this 'if' condition should be as below (psedo code):

if (callback == EXECUTED) {
  idx_op_ctx->cbs->oop_executed = s3_motr_op_executed;
  idx_op_ctx->cbs->oop_stable = Null;
} else if (callback == STABLE) {
  idx_op_ctx->cbs->oop_executed = Null;
  idx_op_ctx->cbs->oop_stable = s3_motr_op_stable;
} else { some error }

Further, 's3_motr_op_stable' can be same as 's3_motr_op_executed' . Why should S3 server make any difference in the called callback? But, we need to set the other callback as null; otherwise, it would call both callbacks. This is my thought...may be wrong.